lhotse/audio.py

```diff
@@ -1102,3 +1107,44 @@ def parse_channel_from_ffmpeg_output(ffmpeg_stderr: bytes) -> str:
         f"Could not determine the number of channels for OPUS file from the following ffmpeg output "
         f"(shown as bytestring due to avoid possible encoding issues):\n{str(ffmpeg_stderr)}"
     )
+
+
+def sph_info(path: Pathlike) -> LibsndfileCompatibleAudioInfo:
+    samples, sampling_rate = read_sph(path)
+    return LibsndfileCompatibleAudioInfo(
+        channels=samples.shape[0],
+        frames=samples.shape[1],
+        samplerate=sampling_rate,
+        duration=samples.shape[1] / sampling_rate
+    )
+
+
+def read_sph(
+        sph_path: Pathlike,
+        offset: Seconds = 0.0,
+        duration: Optional[Seconds] = None
+) -> Tuple[np.ndarray, int]:
+    """
+    Reads SPH files using sph2pipe in a shell subprocess.
+    Unlike audioread, correctly supports offsets and durations for reading short chunks.
+
+    :return: a tuple of audio samples and the sampling rate.
+    """
+
+    sph_path = Path(sph_path)
+
+    # Construct the sph2pipe command depending on the arguments passed.
+    cmd = f'sph2pipe -f wav -p -t {offset}:'
+
+    if duration is not None:
+        cmd += f'{round(offset + duration, 5)}'
+    # Add the input specifier after offset and duration.
+    cmd += f' {sph_path}'
+
+    # Actual audio reading.
+    proc = BytesIO(run(cmd, shell=True, stdout=PIPE, stderr=PIPE).stdout)
```

Comment:
This makes sure the failures are propagated. 

```suggestion
    proc = BytesIO(run(cmd, shell=True, check=True, stdout=PIPE, stderr=PIPE).stdout)
```